frontend/src/Connection.ts

```diff
@@ -63,8 +63,11 @@ export class Connection {
         clean();
         resolve(null);
       }
-
-      const socket = new WebSocket(Connection.address);
+      const address = localStorage.getItem('connectionURI');
```

Comment:
Where is this local storage value set? Is this for only for mock connetion injection? This should almost definitely be moved to line 30 along other checks (or the whole thing should be moved to the constructor if you don't want to nest ternarny expressions).